osu.Game/Screens/Edit/Components/Timelines/Summary/Parts/GroupVisualisation.cs

```diff
@@ -60,5 +57,11 @@ protected override void LoadComplete()
                 }
             }, true);
         }
+
+        /// <summary>
+        /// For display purposes, check whether the proposed group is made redundant by this visualisation group.
+        /// </summary>
+        public bool IsVisuallyRedundant(ControlPointGroup other) =>
+            other.ControlPoints.Any(c => InternalChildren.OfType<IControlPointVisualisation>().Any(c2 => c2.IsVisuallyRedundant(c)));
```

Comment:
Not `.All(c => InternalChildren...)`?

Reply:
If you think about it, this is matching against all surrounding control points. So there could be a (matching) point which makes this one redundant, but then a completely unrelated point which needs to be rendered.

This check is looking for one match which signals that we can be omitted. If matching against `All` there would be a considerable number of false negatives.

Bit confusing logic, I'll give you that. Maybe the above needs to be posted inline?

Reply:
> So there could be a (matching) point which makes this one redundant, but then a completely unrelated point which needs to be rendered.

> If matching against `All` there would be a considerable number of false negatives.

Aren't these two statements in conflict with each other? If both groups have the same timing point but different difficulty points then that group won't be displayed, if I'm understanding this logic correctly.

Reply:
Ah, so you're specifically talking about the group condition. I think you're right on that one, but I wrote it this way because it would be such a rare scenario.

Let me try changing and make sure it still works as expected.

Reply:
Looks to be more correct. I've made the change.